DataProcessing/ExtractAlphaTrueBeatsConverter.cs

```diff
@@ -0,0 +1,460 @@
+        }
+
+        /// <summary>
+        /// Writes processed data to disk, merging any existing data with the newly processed data.
+        /// </summary>
+        /// <param name="processingData">The data to write to disk</param>
+        /// <param name="processingDate">Processing date</param>
+        protected void WriteToFile(Dictionary<string, List<ExtractAlphaTrueBeat>> processingData, DateTime processingDate)
+        {
+            foreach (var kvp in processingData)
+            {
+                var ticker = kvp.Key;
+                var trueBeatData = kvp.Value
+                    .Where(x => x.Time.Date == processingDate)
+                    .OrderBy(x => x.FiscalPeriod.FiscalYear)
+                    .ThenBy(x => x.FiscalPeriod.FiscalQuarter ?? 0)
+                    .ThenBy(x => x.EarningsMetric);
+
+                var outputData = new List<ExtractAlphaTrueBeat>();
+                var processedData = new FileInfo(
+                    Path.Combine(
+                        ExistingDataDirectory.FullName,
+                        "alternative",
+                        "extractalpha",
+                        "truebeats",
+                        $"{ticker.ToLowerInvariant()}.csv"));
+
+                if (processedData.Exists)
+                {
+                    outputData = File.ReadAllLines(processedData.FullName)
+                        .Select(x => (ExtractAlphaTrueBeat) _factory.Reader(_config, x, processingDate, false))
+                        .OrderBy(x => x.Time)
+                        .ThenBy(x => x.FiscalPeriod.FiscalYear)
+                        .ThenBy(x => x.FiscalPeriod.FiscalQuarter ?? 0)
+                        .ThenBy(x => x.EarningsMetric)
+                        .ToList();
+                }
+
+                var outputDataLines = outputData
+                    .Select(ToCsv)
+                    .ToList();
+                   
+                var existingDataLinesSet = outputDataLines.ToHashSet();
+
+                foreach (var trueBeat in trueBeatData)
+                {
+                    var csvLine = ToCsv(trueBeat);
+                    if (existingDataLinesSet.Add(csvLine))
+                    {
+                        outputDataLines.Add(csvLine);
+                    }
+                }
+
+                var outputFileDirectory = Directory.CreateDirectory(
+                    Path.Combine(
+                        OutputDataDirectory.FullName,
+                        "alternative",
+                        "extractalpha",
+                        "truebeats"));
+                
+                var outputFilePath = Path.Combine(
+                    outputFileDirectory.FullName, 
+                    $"{ticker.ToLowerInvariant()}.csv");
+                
+                Log.Trace($"ExtractAlphaTrueBeatsConverter.WriteToFile(): Writing data to: {outputFilePath}");
+                File.WriteAllText(outputFilePath, string.Join("\n", outputDataLines));
```

Comment:
Hm `outputDataLines` being a hash set doesn't guarantee the order, don't think this is right? Should probably just override existing data here?